File: app/src/main/java/in/aerem/comconbeacons/PositionsWebService.java
package in.aerem.comconbeacons;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

import java.util.List;

// See Swagger API documentation at http://85.143.222.113/api/documentation
public interface PositionsWebService {
    @POST("login")
    Call<LoginResponse> login(@Body LoginRequest request);

    @POST("register")
    Call<LoginResponse> register(@Body LoginRequest request);

    @POST("positions")
    Call<PositionsResponse> positions(@Header("Authorization") String authorization, @Body PositionsRequest request);

    @GET("users")
    Call<List<UsersResponse>> users();
}

File: app/src/main/java/in/aerem/comconbeacons/PositionsWebService.kt
package `in`.aerem.comconbeacons

import `in`.aerem.comconbeacons.models.*
import retrofit2.Call
import retrofit2.http.*

// See Swagger API documentation at http://85.143.222.113/api/documentation
interface PositionsWebService {
    @POST("login")
    fun login(@Body request: LoginRequest): Call<LoginResponse>

    @POST("register")
    fun register(@Body request: RegisterRequest): Call<LoginResponse>

    @POST("positions")
    fun positions(@Header("Authorization") authorization: String, @Body request: PositionsRequest): Call<PositionsResponse>

    @GET("users")
    fun users(): Call<List<UserResponse>>

    @PUT("profile")
    fun profile(@Header("Authorization") authorization: String, @Body request: ProfileRequest): Call<UserResponse>
}

